hogben/simulate.py

```diff
@@ -10,368 +10,270 @@
+        probe.dq = self.dq
+
+        # Adjust probe calculation for constant dQ/Q resolution.
+        argmin, argmax = np.argmin(q_array), np.argmax(q_array)
+        probe.calc_Qo = np.linspace(q_array[argmin] - 3.5 * dq_array[argmin],
+                                    q_array[argmax] + 3.5 * dq_array[argmax],
+                                    21 * len(q_array))
+
+        # If the sample is magnetic, create a polarised QProbe.
+        if self.sample_model.ismagnetic:
+            probes = [None] * 4
+            probes[spin_state] = probe
+            probe = refl1d.probe.PolarizedQProbe(xs=probes, name='')
+            probe.spin_state = spin_state
+
+        return refl1d.experiment.Experiment(probe=probe, sample=self.sample_model)
+
+    def reflectivity(self, q: np.ndarray) -> np.ndarray:
+        """Calculates the model reflectivity at given `q` points.
+
+        Args:
+            q: Q points to calculate reflectance at.
+
+        Returns:
+            numpy.ndarray: reflectivity for each Q point.
+
+        """
+        # If there are no data points, return an empty array.
+        if len(q) == 0:
+            return np.array([])
+
+        # Calculate the reflectance in either refnx or Refl1D.
+        if isinstance(self.sample_model, refnx.reflect.Stucture):
+            self.model = refnx.reflect.ReflectModel(self.sample_model,
+                                                    scale=self.scale,
+                                                    bkg=self.bkg,
+                                                    dq=self.dq)
+            return self.model(q)
+
+        if isinstance(self.sample_model, refl1d.model.Stack):
+            # If magnetic, use the correct spin state.
+            experiment = self.refl1d_experiment(q,
+                                                self.sample_model.probe.spin_state)
+
+            if self.sample_model.ismagnetic:
+                return experiment.reflectivity()[self.sample_model.probe.spin_state][1]
+            # experiment.reflectivity() returns q, r, or an array if magnetic
+
+            else:
+                return experiment.reflectivity()[1]
+
+    def _run_experiment(self, angle: float, points: int, time: float,
+                        spin_state: Optional[str] = None) -> tuple:
+        """Simulates a single angle measurement of a given 'model' on the
+        instrument set in self.incident_flux_data with the defined spin state,
+        unpolarised measurements (the default) have the set to None
+
+        Args:
+            angle: angle to simulate.
+            points: number of points to use for simulated data.
+            time: counting time for simulation.
+            spin_state: spin state to simulate if given a magnetic sample.
+
+        Returns:
+            tuple: simulated Q, R, dR data and incident neutron counts.
+
+        """
+        # direct_beam = [wavelength, flux]
+        wavelengths, flux = self.incident_flux_data
```

Comment:
For `OFFSPEC_non_polarised_old.dat` (which seems to be the default direct beam file),  `self.incident_flux_data` returns an array with shape (2044, 3). So there's two issues with this line:

1 - There's three columns, so you will need to assign the data to three different variables if you want to unpack it like this 
2 - It unpacks the other way around, for it to work it should be transposed as well: `wavelengths, flux = np.transpose(self.incident_flux_data)`. However, then there's still the third column to be unpacked. The  `OFFSPEC_non_polarised.dat` data-file is two-column, so there it should work without a third variable.

Perhaps the previous way (`wavelength = self.incident_flux_data[:, 0]` and `flux = self.incident_flux_data[:, 1]`) is slightly more robust, albeit a bit less clean perhaps.